diff -Naur etc.orig/rc.d/rc.fsck etc/rc.d/rc.fsck
--- etc.orig/rc.d/rc.fsck	2006-05-30 14:00:14.000000000 +0200
+++ etc/rc.d/rc.fsck	2007-01-03 13:39:21.000000000 +0100
@@ -30,7 +30,7 @@
 	if [ ! $rw = yes ]; then
 		if [ ! -r /etc/fastboot ]; then
 			start "$fscheck"
-			/sbin/fsck $ffsck -C -a -T /
+			/sbin/fsck $ffsck -C -a -T / >/dev/null 2>&1
 			fsstate=$?
 			ok $?
 		else
@@ -53,14 +53,14 @@
 			fi
 			echo "$umountfs"
 			umount -a -r
-			mount -n -o remount,ro /
+			mount -n -o remount,ro / >/dev/null 2>&1
 			echo "$rebootsys"
 			sleep 2
 			reboot -f
 		fi
 		# Remount the root filesystem in read-write mode
 		start "$remountrw"
-		mount -w -n -o remount /
+		mount -w -n -o remount / >/dev/null 2>&1
 		if [ $? -gt 0 ] ; then
 			ok 1
 			echo -e "$failedrw"
@@ -78,7 +78,7 @@
 	# Any /etc/mtab that exists here is old, so we delete it to start over:
 	/bin/rm -f /etc/mtab*
 	# Remounting the / partition will initialize the new /etc/mtab:
-	mount -w -o remount /
+	mount -w -o remount / >/dev/null 2>&1
 	
 	# Fix /etc/mtab to list sys and proc if they were not yet entered in
 	# /etc/mtab because / was still mounted read-only:
diff -Naur etc.orig/rc.d/rc.mount etc/rc.d/rc.mount
--- etc.orig/rc.d/rc.mount	2006-12-13 10:52:25.000000000 +0100
+++ etc/rc.d/rc.mount	2007-01-03 13:40:20.000000000 +0100
@@ -23,6 +23,11 @@
 
 rc_start()
 {
+	# starting fstab auto updater
+	if [ -e /usr/local/bin/fstab-update ]; then
+	    /usr/local/bin/fstab-update --daemon &
+	fi
+
 	# See if a forced filesystem check was requested at shutdown.
 	if [ -r /etc/forcefsck ]; then
 		ffsck="-f"
@@ -31,14 +36,14 @@
 	# Check all the non-root filesystems.
 	if ! [ -r /etc/fastboot ]; then
 		msg $"Checking non-root filesystems"
-		/sbin/fsck $ffsck -T -C -R -A -a
+		/sbin/fsck $ffsck -T -C -R -A -a >/dev/null 2>&1
 		ok $?
 	fi
 
 	# Mount non-root file systems in fstab (but not NFS or SMB 
 	# because TCP/IP is not yet configured).
 	msg $"Mounting local filesystems"
-	mount -a -t nonfs,nosmbfs,noproc,nosysfs,nofuse
+	mount -a -t nonfs,nosmbfs,noproc,nosysfs,nofuse,tmpfs >/dev/null 2>&1
 	ok $?
 
 	# Clean up some temporary files.
@@ -55,7 +60,7 @@
 	if [ "$rfstype" = "umsdos" ]; then
 		# We need to update any files added in DOS.
 		msg $"Synchronizing UMSDOS directory structure"
-		umssync -r99 -v- /
+		umssync -r99 -v- / >/dev/null 2>&1
 		ok $?
 	fi
 }
@@ -64,15 +69,8 @@
 {
 	if (( ${rl} == 0 )) || (( ${rl} == 6 )) ; then
 		msg $"Unmounting local filesystems"
-		umount -a -r -t nonfs,noumsdos,nosmbfs,noproc,nofuse
+		umount -a -r -t nonfs,noumsdos,nosmbfs,noproc,nofuse,tmpfs >/dev/null 2>&1
 		ok $?
-		
-		# Don't remount UMSDOS root volumes.
-		if [ ! "`mount | head -n 1 | cut -d ' ' -f 5`" = "umsdos" ]; then
-			msg $"Remounting root filesystem read-only"
-			mount -n -o remount,ro /
-			ok $?
-		fi
 	fi
 	# This never hurts. :)
 	sync
