diff -Naur etc.old/rc.d/rc.reboot etc/rc.d/rc.reboot
--- etc.old/rc.d/rc.reboot	2005-11-27 22:37:21.000000000 +0100
+++ etc/rc.d/rc.reboot	2006-04-18 23:30:24.000000000 +0200
@@ -22,13 +22,16 @@
 
 rc_start()
 {
+	sysctl -w dev.cdrom.lock=0 >/dev/null
+	sysctl -w dev.cdrom.autoeject=1 >/dev/null
+	sysctl -w dev.cdrom.autoclose=1 >/dev/null
+	
 	if (( ${rl} == 0 )) ; then
-		poweroff
+		echo -n "Now you can switch your computer off..."
 	else
-		msg $"Rebooting the system"
-		reboot
-		ok $?
+		echo -n "Now you can reboot your system..."
 	fi
+	eject -p -m cdrom
 }
 
 rc_exec $1
