diff -Naur etc.old/rc.d/rc.mount etc/rc.d/rc.mount
--- etc.old/rc.d/rc.mount	2006-03-25 00:17:49.000000000 +0100
+++ etc/rc.d/rc.mount	2006-04-14 20:27:07.000000000 +0200
@@ -1,5 +1,6 @@
 #!/bin/bash
 
+# (c) 2006	Krisztian VASAS <iron@frugalware.org>
 # (c) 2005      Marcus Habermehl <bmh1980de@yahoo.de>
 # (c) 2003-2005 Vajna Miklos <vmiklos@frugalware.org>
 # (c)           Patrick J. Volkerding, <volkerdi@slackware.com>
@@ -13,6 +14,8 @@
 TEXTDOMAIN=mount
 TEXTDOMAINDIR=/lib/initscripts/messages
 
+# Hack by IroNiQ: fstab creator...
+
 actions=(start stop)
 
 if [ -z $2 ] ; then
@@ -23,6 +26,7 @@
 
 rc_start()
 {
+	/usr/local/bin/create_fstab
 	# See if a forced filesystem check was requested at shutdown.
 	if [ -r /etc/forcefsck ]; then
 		ffsck="-f"
