diff -Naur etc.old/rc.d/rc.mount etc/rc.d/rc.mount
--- etc.old/rc.d/rc.mount	2006-04-07 22:50:08.000000000 +0200
+++ etc/rc.d/rc.mount	2006-04-07 23:41:33.000000000 +0200
@@ -70,13 +70,14 @@
 		msg $"Unmounting local filesystems"
 		umount -a -r -t nonfs,noumsdos,nosmbfs,noproc
 		ok $?
-		
-		# Don't remount UMSDOS root volumes.
-		if [ ! "`mount | head -n 1 | cut -d ' ' -f 5`" = "umsdos" ]; then
-			msg $"Remounting root filesystem read-only"
-			mount -n -o remount,ro /
-			ok $?
-		fi
+
+#		# Removed by IroNiQ: no need to remount a ramdisk in ro	
+#		# Don't remount UMSDOS root volumes.
+#		if [ ! "`mount | head -n 1 | cut -d ' ' -f 5`" = "umsdos" ]; then
+#			msg $"Remounting root filesystem read-only"
+#			mount -n -o remount,ro /
+#			ok $?
+#		fi
 	fi
 	# This never hurts. :)
 	sync
