diff -Naur etc.old/rc.d/rc.dbus etc/rc.d/rc.dbus
--- etc.old/rc.d/rc.dbus	2006-02-25 15:49:47.000000000 +0100
+++ etc/rc.d/rc.dbus	2006-04-02 11:08:14.000000000 +0200
@@ -21,6 +21,6 @@
 	"$0" start
 else # start
 	start "$startdbus"
-	/usr/bin/dbus-daemon --system
+	/usr/bin/dbus-daemon --system >/dev/null 2>&1
 	ok $?
 fi
diff -Naur etc.old/rc.d/rc.hald etc/rc.d/rc.hald
--- etc.old/rc.d/rc.hald	2006-02-25 19:34:28.000000000 +0100
+++ etc/rc.d/rc.hald	2006-04-02 11:08:14.000000000 +0200
@@ -23,7 +23,7 @@
 else # start
 	start "$starthald"
 	if [ -z "$PID" ]; then
-		/usr/sbin/hald
+		/usr/sbin/hald >/dev/null 2>&1
 	fi
 	ok $?
 fi
diff -Naur etc.old/rc.d/rc.modules etc/rc.d/rc.modules
--- etc.old/rc.d/rc.modules	2006-02-23 18:31:27.000000000 +0100
+++ etc/rc.d/rc.modules	2006-04-02 11:08:14.000000000 +0200
@@ -40,7 +40,7 @@
 		# only rebuild dependencies if new module(s) are found:
 		if ! [ "" = "$news" ]; then
 			start "$updatemoddeps"
-			/sbin/depmod -a
+			/sbin/depmod -a >/dev/null 2>&1
 			ok $?
 		else
 			start "$noupdate"
@@ -50,7 +50,7 @@
 		# we don't have find, or there is no existing modules.dep,
 		# or it is out of date.
 		start "$updatemoddeps"
-		/sbin/depmod -A
+		/sbin/depmod -A >/dev/null 2>&1
 		ok $?
 	fi
 
@@ -59,7 +59,7 @@
 	## At here you can do anything, run any commands, example iptables or any other base
 	## commands like in the older rc.modules file.
 	if grep -q quota /etc/fstab ; then
-	    $MODPROBEBIN quota_v2
+	    $MODPROBEBIN quota_v2 >/dev/null 2>&1
 	fi
 
 	## Starting reading conf file and loading modules
@@ -81,10 +81,10 @@
   				## If yes:  			
   				arg=$(echo $module | cut -f2 -d '"')
   				module=$(echo $module | cut -f1 -d '"')
-  				$MODPROBEBIN $module $arg
+  				$MODPROBEBIN $module $arg >/dev/null 2>&1
   			else
   				## If not simply modprobe it
-  				$MODPROBEBIN $module
+  				$MODPROBEBIN $module >/dev/null 2>&1
   			fi
   			fi
 
diff -Naur etc.old/rc.d/rc.mount etc/rc.d/rc.mount
--- etc.old/rc.d/rc.mount	2006-04-01 10:27:32.000000000 +0200
+++ etc/rc.d/rc.mount	2006-04-02 11:08:14.000000000 +0200
@@ -42,7 +42,7 @@
 	# Mount non-root file systems in fstab (but not NFS or SMB 
 	# because TCP/IP is not yet configured).
 	msg $"Mounting local filesystems"
-	mount -a -t nonfs,nosmbfs,noproc,nosysfs
+	mount -a -t nonfs,nosmbfs,noproc,nosysfs 2>&1 >/dev/null
 	ok $?
 
 	# Clean up some temporary files.
@@ -59,7 +59,7 @@
 	if [ "$rfstype" = "umsdos" ]; then
 		# We need to update any files added in DOS.
 		mgs $"Synchronizing UMSDOS directory structure"
-		umssync -r99 -v- /
+		umssync -r99 -v- / >/dev/null 2>&1
 		ok $?
 	fi
 }
diff -Naur etc.old/rc.d/rc.random etc/rc.d/rc.random
--- etc.old/rc.d/rc.random	2005-12-29 22:32:35.000000000 +0100
+++ etc/rc.d/rc.random	2006-04-02 11:08:33.000000000 +0200
@@ -46,7 +46,7 @@
 		rm -f /tmp/hunt*
 	fi
 	chmod 755 / 2> /dev/null
-	chmod 1777 /tmp /var/tmp
+	chmod 1777 /tmp /var/tmp 2>/dev/null
 
 	if [ -f /etc/random-seed ]; then
 		cat /etc/random-seed >/dev/urandom
diff -Naur etc.old/rc.d/rc.udev etc/rc.d/rc.udev
--- etc.old/rc.d/rc.udev	2006-03-24 19:10:01.000000000 +0100
+++ etc/rc.d/rc.udev	2006-04-02 11:08:14.000000000 +0200
@@ -11,7 +11,7 @@
 # Thanks go out to the Gentoo developers for proving 
 # that this is possible to do.
 
-mount / -o rw,remount
+#mount / -o rw,remount
 mount none -t sysfs /sys
 
 . /etc/udev/udev.conf
@@ -76,7 +76,7 @@
 fi
 
 #echo "mounting... ramfs at $udev_root"
-mount -n -t ramfs none $udev_root
+mount -n -t ramfs none $udev_root >/dev/null 2>&1
 
 # propogate /udev from /sys
 export ACTION=add
@@ -95,12 +95,12 @@
 #echo "udev startup is finished!"
 
 if ls /lib/udev/devices/* 2>/dev/null >/dev/null;then
-	cp -a /lib/udev/devices/* /dev
+	cp -a /lib/udev/devices/* /dev  >/dev/null 2>&1
 fi
 
 # check if udevd is already running
 if ! [ `pidof -o %PPID /sbin/udevd` ]; then
-	/sbin/udevd --daemon
+	/sbin/udevd --daemon >/dev/null 2>&1
 fi
 
 # Make udevsend the hotplug multiplexer
@@ -110,6 +110,6 @@
 else
 	echo "/sbin/udevsend" > /proc/sys/kernel/hotplug
 fi 
-mount / -o ro,remount
+mount / -o ro,remount >/dev/null 2>&1
 
 # end of rc.udev
