diff -Naur etc.orig/rc.d/rc.reboot etc/rc.d/rc.reboot
--- etc.orig/rc.d/rc.reboot	2006-10-20 00:13:45.000000000 +0200
+++ etc/rc.d/rc.reboot	2006-11-09 21:56:00.000000000 +0100
@@ -11,24 +11,45 @@
 source /lib/initscripts/functions
 TEXTDOMAIN=reboot
 TEXTDOMAINDIR=/lib/initscripts/messages
+. /usr/local/bin/liblinuxlive
 
 actions=(start)
 
 if [ -z $2 ] ; then
-	rl=4
+    rl=4
 else
-	rl=$2
+    rl=$2
 fi
 
 rc_start()
 {
-	if (( ${rl} == 0 )) ; then
-		poweroff
-	else
-		msg $"Rebooting the system"
-		reboot
-		ok $?
-	fi
+# eject cdrom devices
+CD=`ls -1 /mnt 2>/dev/null | grep cdrom | sed 's/_.*//'`
+SLEEP=10
+if [ ! "`cmdline_parameter copy2ram`" ]; then
+  if [ "$CD" ]; then
+  echo "Ejecting CD-ROM..."
+	for i in $CD ; do
+       		/usr/bin/eject -m /dev/$i >/dev/null 2>&1
+	done
+	until [ $SLEEP -lt 0 ] ; do		
+		echo -ne "Closing CD-ROM... $SLEEP s    	\r"
+		let SLEEP-=1
+		sleep 1
+	done
+  echo "Closing CD-ROM...         "
+	for i in $CD ; do
+		/usr/bin/eject -t /dev/$i >/dev/null 2>&1
+	done
+  fi
+fi
+if (( ${rl} == 0 )) ; then
+    poweroff
+else
+    msg $"Rebooting the system"
+    reboot
+    ok $?
+fi
 }
 
 rc_exec $1
