diff -Naur var.orig/lib/frugalware/system/service var/lib/frugalware/system/service
--- var.orig/lib/frugalware/system/service      2008-01-27 08:39:02.000000000 +0100
+++ var/lib/frugalware/system/service   2008-01-27 08:43:20.000000000 +0100
@@ -30,6 +30,10 @@
 	echo $"del          Removes a service from chkconfig management."
 	echo
 	echo $"list         List whether the service is stopped or started in each runlevel."
+	echo
+	echo $"save         Save config settings. Only in service config."
+	echo
+	echo $"restore      Restore saved settings. Only in service config."
 	exit 0
 fi
 if [ "$1" = "list" ]; then
@@ -60,5 +64,9 @@
 	chkconfig --del $service
 elif [ "$2" = "list" ]; then
 	chkconfig --list $service
+elif [ "$2" = "save" ]; then
+	$name save
+elif [ "$2" = "restore" ]; then
+	$name restore
   fi
 fi
