diff -Naur var.orig/lib/frugalware/system/service var/lib/frugalware/system/service
--- var.orig/lib/frugalware/system/service	2006-11-07 07:29:41.000000000 +0100
+++ var/lib/frugalware/system/service	2006-11-07 07:27:30.000000000 +0100
@@ -35,6 +35,10 @@
 	chkconfig --del $service
 elif [ "$2" = "list" ]; then
 	chkconfig --list $service
+elif [ "$2" = "save" ]; then
+	$name save
+elif [ "$2" = "restore" ]; then
+	$name restore
 else # help
 	echo "service 0.1 (C) 2003-2004 Vajna Miklos"
 	echo
@@ -47,5 +51,7 @@
 	echo $"add          Adds a new service for management by chkconfig."
 	echo $"del          Removes a service from chkconfig management."
 	echo $"list         List whether the service is stopped or started in each runlevel."
+	echo $"save         Save config settings. Only in service config."
+	echo $"restore      Restore saved settings. Only in service config."
   fi
 fi
